### ISSUE TITLE
JSON File Pointer resolving behaviour

### ISSUE THREAD

**Issue by marcortw** [42Crunch/vscode-openapi#113]
I am not even a 100% sure if I am at the right place, here, but it seems like this extension is the one which magically resolves some json pointers - but not all that I would wish for :-) I am working with an asyncapi document and do not get JSON pointers resolved which are within a reference by file. I think this used to work in the past. here's a minimal example:

minimal_start.yml
```yaml
asyncapi: 2.0.0
info:
  title: Hello world application
  version: '0.1.0'
channels:
  hello:
    publish:
      message:
        payload:
          type: string
          pattern: '^hello .+$'
  goodbye:
    subscribe:
      message:
        payload:
          oneOf:
            - $ref: './goodbye_direct.json'
            - $ref: './goodbye_indirect.json#/goodbye' 
```
goodbye_direct.json:
```json
{
  "name": "directly linked goodbye payload",
  "type": "object",
  "properties": {
    "message": {
      "type": "string",
      "const": "goodbye"
    }
  }
}
```

goodbye_indirect.json:
```json
{
  "goodbye": {
    "name": "indirectly linked goodbye payload",
    "type": "object",
    "properties": {
      "message": {
        "type": "string",
        "const": "goodbye"
      }
    }
  }
}
```

I believe this is all valid JSON pointers etc. - the parser/generator i'm using doesn't complain (https://github.com/asyncapi/generator). What I experience is the following.

For the direct reference, vs code resolves the link/definition (here done with ctrl-and-hovering over the reference) and lets me navigate directly to it (which is extremely useful if you're constructing slightly more complex schemas):
![image](https://user-images.githubusercontent.com/6734691/109316712-31021600-784c-11eb-8da6-0de2c78ac78d.png)

For the indirect reference, this doesn't work:
![image](https://user-images.githubusercontent.com/6734691/109317030-8b02db80-784c-11eb-9e2e-13e7c54e4e53.png)

Sorry I accidentally submitted that issue too early, now updating with all the infos i think are relevant.

**Comment by ak1394** [comment]
Hi @marcortw ! Yes, if it have worked for you before you might have been using a different extension. This one does not support asyncapi yet https://github.com/42Crunch/vscode-openapi/issues/68

**Comment by marcortw** [comment]
@ak1394 , see my updated issue text - isn't that done by this extension?

**Comment by ak1394** [comment]
Well, there's been some refactoring in v4 which might have changed how it works with files which are not officially supported "openapi" files.  Given that we do spill out some of our functionality into even non-openapi files, I'll have a look to see if I can see fix it.

**Comment by marcortw** [comment]
Thanks for reopening! I quickly did a test with an OAS yaml file using the same style, and the behaviour is indeed the same.

**Comment by ak1394** [comment]
Thanks for doing the test! In this case the problem is likely caused by the latest update. I'll look to fix it, but in the mean time you can try installing 3.10 which doesn't have the latest changes.

**Comment by ak1394** [comment]
I've found the issue, it will be resolved in the next version of the extension which should be released pretty soon.

**Comment by ak1394** [comment]
Fixed in 4.0.3

**Comment by marcortw** [comment]
Awesome, thanks for the quick fix!
